crates/rnote-engine/src/pens/selector/penevents.rs

```
@@ -685,6 +686,44 @@ impl Selector {
                             }
                         }
                     }
+                    KeyboardKey::Unicode('x') => {
+                        if let Some(mirror_widget_flags) =
+                            self.mirror(engine_view, MirrorOrientation::Horizontal)
+                        {
+                            widget_flags = mirror_widget_flags;
+
+                            EventResult {
+                                handled: true,
+                                propagate: EventPropagation::Stop,
+                                progress: PenProgress::InProgress,
+                            }
+                        } else {
```

Currently, mirrors that contain text activated through keyboard shortcuts silently fail. I'm not sure how to work around this without access to the `RnAppWindow`.

---

Yeah, you're not going to get access to the `RnAppWindow` inside of the `rnote-engine`. 

If it was a keyboard shortcut activated from an accelerator (see `set_accels_for_action` calls) then it'd go through the same code as clicking the buttons. But then you're not going to be able to add `x` as a shortcut, nor `ctrl+x` (which would even overwrite the `ctrl+x` shortcut when writing 

It's a little different from the rest of the keyboard shortcuts here (as there is no `ctrl` mask contrary to the rest. Though you can't do `ctrl+x` either).

Here the way to communicate to the engine would be through  the `WidgetFlags` (add a field, handle it in the `handle_widget_flags`). Does it make sense to modify the widget flags to accomodate this ? Maybe add a field for an error message ?

Another thing I checked on `WidgetFlags`, I'm not sure it matters much to do a `Option<WidgetFlags>`. It's kinda not used in the code, because handling a default widget flags will by default do nothing. Although maybe it's better in principle to pass around the default widget flags even if you know it does nothing _in case_ another step is added later on that modifies it. I don't feel too strongly about this though.

Though if you add a message field to the `WidgetFlags`, this will obviously change (generic popup in the handle function so the mirror would always return a widget flags)